### PR TITLE
:sparkles: Rebase to v0.17.2

### PR DESCRIPTION
**Pull request by sttts**
Forked v0.17.2 as `kcp-0.17` and cherry-picked the kcp commit from kcp-1.28 branch.

### CODE REVIEW

File: DOWNSTREAM_OWNERS [hunk 1]
```diff
@@ -0,0 +1,8 @@
+approvers:
+  - davidfestal
+  - ncdc
+  - stevekuznetsov
+  - sttts
+reviewers:
+  - fabianvf
+  - varshaprasad96
```
Comment (mjudeikis) [inline DOWNSTREAM_OWNERS]: Still right?

File: pkg/cache/internal/cache_reader.go [hunk 2]
```diff
@@ -253,3 +277,9 @@ func KeyToNamespacedKey(ns string, baseKey string) string {
 	}
 	return allNamespacesNamespace + "/" + baseKey
 }
+
+// KeyToClusteredKey prefixes the given index key with a cluster name
+// for use in field selector indexes.
+func KeyToClusteredKey(clusterName string, ns string, baseKey string) string {
+	return clusterName + "/" + KeyToNamespacedKey(ns, baseKey)
```
Comment (mjudeikis) [inline pkg/cache/internal/cache_reader.go]: Internally we use `cluster|namespace/name`. changing here?

**Comment by mjudeikis** [comment]
/lgtm
Few comments

File: DOWNSTREAM_OWNERS (hunk 1 above)
Comment (sttts) [reply DOWNSTREAM_OWNERS]: good point. Let me clean up.

File: pkg/cache/internal/cache_reader.go (hunk 2 above)
Comment (sttts) [reply pkg/cache/internal/cache_reader.go]: yep, let's use the chance

Comment (sttts) [reply pkg/cache/internal/cache_reader.go]: done

File: DOWNSTREAM_OWNERS (hunk 1 above)
Comment (sttts) [reply DOWNSTREAM_OWNERS]: done

**Pull request merged**